server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -197,12 +207,23 @@ public long getLength() {
 
         public static UploadedSegmentMetadata fromString(String uploadedFilename) {
             String[] values = uploadedFilename.split(SEPARATOR);
-            return new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            UploadedSegmentMetadata metadata = new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            metadata.setWrittenBy(values[4]);
```

Comment:
why not create an overloadeded constructor where we can pass writtenBy as well?

Reply:
I didn't overload the constructor because the same result could be achieved by using the setter (which we currently need anyway). This is also related to the your second question below.

Reply:
Hey @linuxpi, 
Marking this thread as resolved for now because we don't have a better way to do this currently.
Thanks!